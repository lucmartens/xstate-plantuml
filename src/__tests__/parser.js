const parse = require('../parser').default;

describe('parser', () => {
  test('simple machine', () => {
    const machine = {
      key: 'users-crud',
      initial: 'on',
      states: {
        on: {
          on: {
            TOGGLE: 'off'
          }
        },
        off: {
          on: {
            TOGGLE: 'on'
          }
        }
      }
    };

    expect(parse(machine)).toMatchSnapshot();
  });

  test('parallel machine', () => {
    const machine = {
      key: 'users-crud',
      parallel: true,
      states: {
        light: {
          initial: 'on',
          states: {
            on: {
              on: {
                TOGGLE: 'off'
              }
            },
            off: {
              on: {
                TOGGLE: 'on'
              }
            }
          }
        }
      }
    };

    expect(parse(machine)).toMatchSnapshot();
  });
});
