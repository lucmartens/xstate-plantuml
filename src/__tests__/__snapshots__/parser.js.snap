// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser parallel machine 1`] = `
Array [
  "root",
  Object {},
  Array [
    "machine",
    Object {
      "key": "light",
    },
    Array [
      "state",
      Array [
        "on",
        "light_on",
      ],
    ],
    Array [
      "state",
      Array [
        "off",
        "light_off",
      ],
    ],
    Array [
      "event",
      Array [
        "light_on",
        "light_off",
        "TOGGLE",
      ],
    ],
    Array [
      "event",
      Array [
        "light_off",
        "light_on",
        "TOGGLE",
      ],
    ],
  ],
]
`;

exports[`parser simple machine 1`] = `
Array [
  "root",
  Object {},
  Array [
    "machine",
    Object {
      "key": "users-crud",
    },
    Array [
      "state",
      Array [
        "on",
        "users-crud_on",
      ],
    ],
    Array [
      "state",
      Array [
        "off",
        "users-crud_off",
      ],
    ],
    Array [
      "event",
      Array [
        "users-crud_on",
        "users-crud_off",
        "TOGGLE",
      ],
    ],
    Array [
      "event",
      Array [
        "users-crud_off",
        "users-crud_on",
        "TOGGLE",
      ],
    ],
  ],
]
`;
